ctapipe_io_magic/tests/test_stage1.py

```diff
@@ -26,6 +26,38 @@ def test_stage1():
     ])
     assert ret == 0
 
+    parameters = read_table(output, '/dl1/event/telescope/parameters/tel_001')
+    assert len(parameters) == 910
+
+    trigger = read_table(output, '/dl1/event/subarray/trigger')
+
+    event_type_counts = np.bincount(trigger['event_type'])
+
+    # no pedestals expected, should be only physics data
+    assert event_type_counts.sum() == 910
+    assert event_type_counts[EventType.FLATFIELD.value] == 0
+    assert event_type_counts[EventType.SKY_PEDESTAL.value] == 0
+    assert event_type_counts[EventType.SUBARRAY.value] == 910
+
+
+def test_stage1_single_run():
+    """Test the ctapipe process tool can read in MAGIC real data using the event source"""
+    from ctapipe.tools.process import ProcessorTool
+    from ctapipe.core import run_tool
+
+    tool = ProcessorTool()
+    output = str(test_cal_path).replace(".root", ".h5")
+
+    ret = run_tool(tool, argv=[
+        f'--input={test_cal_path}',
+        f'--output={output}',
+        f'--config={str(config)}',
+        "--MAGICEventSource.process_run=false",
+        "--allowed-tels=[1]",
+        "--camera-frame",
+    ])
+    assert ret == 0
+
     parameters = read_table(output, '/dl1/event/telescope/parameters/tel_001')
     assert len(parameters) == 458
```

Comment:
Above it has 910 events in the subarray trigger table, but now there only 458 events in the parameters table.

For ctapipe, it is usually expected that these tables are matching and that events that could not be reconstructed are there but contain invalid value markers for the steps that failed.

Reply:
But they are different tests, the first one takes in two subruns (looking for all the subruns belonging to the same subrun, and in the test data directory there are two subruns), the second test takes in only the first, so the number of events is different. Otherwise the tests would have failed